### DISCUSSION TITLE
Dynamically add mesh to ShadowGenerator

### DISCUSSION THREAD

**Discussion by snippet-nicotine** [brianzinn/react-babylonjs#160] [Q&A]
Hello,

I'm new to react-babylonjs, and I would like to know the best way to add a mesh generated at runtime to a specific shadowGenerator?
For what i saw in the playground examples, the `shadowGenerator.shadowCasters` needs an array of mesh's name, but what if the mesh is created at runtime and can have any random name?
Does mesh have a special property like a `castShadowFrom={"shadowGeneratorName"}` ?

Thank you

**Comment by brianzinn** [comment]
There is a property called `shadowCastChildren` that may work for you that is not obviously documented:
https://github.com/brianzinn/react-babylonjs/blob/master/docs/api.md#shadowgenerator

What you are able to do can be seen in this PG - the IcoSpheres can be shown/hidden and also the model mesh names are unknown:
https://github.dev/brianzinn/react-babylonjs/blob/master/storybook/stories/babylonjs/VR/withVr.stories.js

**Reply by brianzinn** [reply]
if you didn't want to declaratively nest the dynamic meshes as components and have another idea please share - your `castShadowFrom` could work then possibly, but I need to find the lights then and also that is an interesting use case as the shadow casting has a deferred creation (it cannot be created until after the light is created)..

**Comment by snippet-nicotine** [comment]
Thank you @brianzinn,

I tried the `shadowCastChildren`, but the meshes created don't share the same rootNode. 
The example with the ref is very interesting, I think I will go for that in my case.
Meanwhile, I use the `shadowCastersExcluding={[excludedNames}]` as the ones I want to exclude are few and knowns.

Thank you for these great library you share with us, I am refactoring an house editor and the code is so much readable with the declarative way!!!

**Reply by brianzinn** [reply]
Glad you got it working with `shadowCastersExcluding` - that's how the the example above used to work, so I was excluding meshes from the environment:
https://github.com/brianzinn/react-babylonjs/commit/19e96836d81a54d392edb4838eba5084867b9179#diff-4cf018868b62f6fadd55bf8aabebe3a376ee7190b8bd2b32ec58bdfb31053358L61

Thanks for your kind words - is your house editor online?  Would be interesting to see what you are making!
